src/backend/dx11/src/lib.rs

```diff
@@ -3094,11 +3094,29 @@ pub struct InternalBuffer {
     debug_name: Option<String>,
 }
 
+impl InternalBuffer {
+    unsafe fn release_resources(&mut self) {
+        (&*self.raw).Release();
+        self.raw = ptr::null_mut();
+        self.disjoint_cb.take().map(|cb| (&*cb).Release());
```

Comment:
I'm wary of using functional style (`map`) where you express clear imperative steps, like here. You aren't really mapping a value. if this was written as `if let Some(xx) = yy.take() { do_zz() }` it would look nicer, have less sugar, and be easier to debug, at the cost of slight increase in verbosity

Reply:
I was actually surprised there didn't seem to be a `for_each` equivilant for `Option`. I personally like the functional part of it, but I agree that the verb is inaccurate considering what is happening.

That being said, there is definitely the argument that the if-let will have better debug perf. Will keep in mind going forward.

Reply:
I've just had to debug a bunch of `xxx.do()?.map(|| ...).map_err(|| ...)` and realize how wasteful all the sugar is compared to an old good `match` :)